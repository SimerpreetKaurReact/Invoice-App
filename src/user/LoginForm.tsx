import classes from "./login.module.css"
import { useState } from "react"
import { useForm } from "react-hook-form";
import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from "yup";
import { Button, TextField, Typography, Link, Grid, OutlinedInput, InputAdornment, IconButton, FormHelperText, InputLabel } from "@mui/material";
import { Box, display } from "@mui/system";
import { Visibility, VisibilityOff } from '@mui/icons-material';
import CircularProgress from '@mui/material/CircularProgress';



const loginSchema = yup.object({
    email: yup.string().email().required(),
    password: yup.string().min(5).max(16).required(),
}).required();

type LoginValues = yup.InferType<typeof loginSchema>
interface LoginFormProps {
    onLoginRequest: (data: LoginValues) => Promise<void>
    disabled?: boolean
    error?: string
}
const LoginForm = (props: LoginFormProps) => {
    const { register, handleSubmit, watch, formState: { errors, isSubmitting } } = useForm<LoginValues>({ resolver: yupResolver(loginSchema) });
    const [showPassword, setShowPassword] = useState<boolean>(false)

    const handleClickShowPassword = () => {
        setShowPassword(showPassword => !showPassword,
        );
    };

    return (
        <Box m={2}  >
            <Grid container spacing={2} sx={{ height: "100vh" }}>
                <Grid item sm={12} md={6}>

                    <div className={classes.leftScreen}>


                    </div>
                </Grid>
                <Grid item sm={12} md={6} mt={4} justifyContent="center" >
                    {/* isSubmitting--{isSubmitting.toString()} */}
                    { /* "handleSubmit" will validate your inputs before invoking "onSubmit" */}
                    <Typography variant="h3" align="center" >
                        Invoice managment

                    </Typography>
                    <Typography variant="h5" className={classes.title} align="center" >
                        Login to invoicer

                    </Typography>
                    {props.error && <FormHelperText data-test='form-error'>{props.error}</FormHelperText>}
                    <form onSubmit={handleSubmit(props.onLoginRequest)} style={{
                        padding: 10,
                        display: "flex",
                        flexFlow: 'column',
                        gap: "20px"
                    }}>
                        {/* register your input into the hook by invoking the "register" function */}
                        <InputLabel htmlFor="email">Email</InputLabel>
                        <OutlinedInput defaultValue="" {...register("email")}
                            error={!!errors.email}
                            data-test="email"
                            style={{ width: "100%", background: "#ffddbb78" }}
                        />
                        {/* include validation with required or other stand√∞rd HTML validation rules */}
                        {errors.email?.message && <FormHelperText id="email" data-test="email-error">{errors.email?.message}</FormHelperText>}
                        <InputLabel htmlFor="password">Password</InputLabel>
                        <OutlinedInput {...register("password", { required: true })}
                            id="password"
                            data-test="password"
                            error={!!errors.password}
                            type={showPassword ? 'text' : 'password'}
                            style={{ width: "100%", background: "#ffddbb78" }}
                            endAdornment={
                                <InputAdornment position="end">
                                    <IconButton
                                        aria-label="toggle password visibility"
                                        onClick={handleClickShowPassword}

                                        edge="end"
                                    >
                                        {showPassword ? <VisibilityOff /> : <Visibility />}
                                    </IconButton>
                                </InputAdornment>
                            }
                        />
                        {errors.password?.message && <FormHelperText id="password" data-test="password-error">{errors.password?.message}</FormHelperText>}
                        {/* errors will return when field validation fails  */}


                        <Button variant="contained"
                            data-test='submit-login'
                            type="submit"
                            style={{ background: "#ef9334" }}
                            disabled={props.disabled}>
                            {props.disabled ? <CircularProgress /> : "submit"}
                        </Button>
                    </form>

                    <Typography style={{ fontSize: "1.2rem", color: "darkslategray" }} align="center">
                        New to Invoicer?
                        <Link href="/signup" underline="none">
                            {"Sign up"}
                        </Link>
                    </Typography>

                </Grid >
            </Grid >
        </Box>
    )

}

export default LoginForm
//add enque snackbars for login success
