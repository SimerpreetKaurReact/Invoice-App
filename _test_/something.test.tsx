import { render, screen } from '@testing-library/react'
import LoginForm from '../src/user/LoginForm'
import '@testing-library/jest-dom'

describe('LoginForm', () => {
  it('renders properly', () => {
    render(<LoginForm onLoginRequest={async (values) => { }} />)

    const emailLabel = screen.getByText("email")
    expect(emailLabel).toBeInTheDocument()
  })
  it('snapshots properly', () => {
    const { container } = render(<LoginForm onLoginRequest={async (values) => { }} />)

    expect(container).toMatchSnapshot()
  })
})