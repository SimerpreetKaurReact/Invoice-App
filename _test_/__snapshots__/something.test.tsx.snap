// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshots properly 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 css-6dzwqk-MuiGrid-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Invoice managment
        </h2>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-6 css-10bhsqq-MuiGrid-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Login to invoicer
        </h2>
        <form
          style="padding: 10px; display: flex; flex-flow: column; gap: 20px;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            for="email"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              name="email"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
            for="password"
          >
            Password
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id="password"
              name="password"
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            submit
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
        <h2
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          New to Invoicer?
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1cmksfb-MuiTypography-root-MuiLink-root"
            href="/signup"
          >
            Sign up
          </a>
        </h2>
        ;
      </div>
    </div>
  </div>
</div>
`;
